dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java

```diff
@@ -223,15 +225,20 @@ public Workflow getWorkflow(String repositoryId, Optional<Workflow> existingWork
             workflow.setDescriptorType(DescriptorLanguage.CWL);
             return workflow;
         }
+
+        if (Objects.equals(existingWorkflow.get().getMode(), WorkflowMode.DOCKSTORE_YML)) {
+            String msg = "Cannot refresh .dockstore.yml workflows";
```

Comment:
I don't understand the whole code flow here, but it is confusing that we have a method named `getWorkflow`, whose JavaDoc says it creates or updates a workflow, and whose error message says it cannot refresh the workflow. :)

This may not be easily actionable, but just wanted to note it.

Reply:
Ya I agree it is a bit strange. Get workflow is actually only called on refresh all/individual and will either add a new workflow or update an existing one based on the GitHub repository. I'll be updating this area very soon while working on individual version refresh, so I will make sure to update the function names/descriptions.